src/__tests__/addDefaultProps.coffee

```diff
@@ -0,0 +1,23 @@
+### eslint-disable no-console ###
+import 'jest-dom/extend-expect'
+
+import addDefaultProps from '../addDefaultProps'
+
+describe 'addDefaultProps', ->
+  test 'adds prop from function when missing', ->
+    expect(addDefaultProps(({a}) -> b: a + 1) a: 1).toEqual a: 1, b: 2
```

Comment:
Similar to comment re example style, I'd prefer if the tests mimicked the existing test style more and exercised this helper as part of an actual component

That way it's a little more "integration test"-y than "unit test"-y and provides more confidence in its real-world behavior